File: Jal/Al/src/Chapter1_Fundamentals/Unit2_DataAbstraction/Accumulator.java
package Chapter1_Fundamentals.Unit2_DataAbstraction;

public class Accumulator
{
    private double total;
    private int N;

    public void addDataValue(double val)
    {
        N++;
        total += val;
    }

    public double mean()
    {
        return total/N;
    }

    public String toString()
    {
        return "Mean (" + N + " values): " + String.format("%7.5f", mean());
    }

    public static void main(String[] args) {
        int T = 4;
        Accumulator a = new Accumulator();
        for(int t = 0; t < T; t++)
            a.addDataValue(t);
        StdLib.StdOut.println(a);
    }

}

File: Cal/Al/src/C1/Unit2_DataAbstraction/Accumulator.cpp
#include <string>
#include <sstream>
#include "C1/Unit2_DataAbstraction/Accumulator.h"

using namespace std;

Accumulator::Accumulator()
{
    N = 0;
    total = 0;
}

void Accumulator::addDataValue(double val)
{
    N++;
    total += val;
}

double Accumulator::mean()
{
    return total/N;
}

string Accumulator::toString()
{
    ostringstream outString;
    outString.precision(5);
    outString.width(7);
    outString << this->mean();
    return "Mean (" + to_string(N) + " values): " + outString.str();
}
